resources/dax_clusters.go

```diff
@@ -0,0 +1,293 @@
+						Type:        schema.TypeInt,
+						Resolver:    schema.PathResolver("Endpoint.Port"),
+					},
+					{
+						Name:        "endpoint_url",
+						Description: "The URL that applications should use to connect to the endpoint",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("Endpoint.URL"),
+					},
+					{
+						Name:        "node_create_time",
+						Description: "The date and time (in UNIX epoch format) when the node was launched.",
+						Type:        schema.TypeTimestamp,
+					},
+					{
+						Name:        "node_id",
+						Description: "A system-generated identifier for the node.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "node_status",
+						Description: "The current status of the node",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "parameter_group_status",
+						Description: "The status of the parameter group associated with this node",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+func fetchDaxClusters(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	c := meta.(*client.Client)
+	svc := c.Services().DAX
+
+	config := dax.DescribeClustersInput{}
+	for {
+		output, err := svc.DescribeClusters(ctx, &config, func(o *dax.Options) {
+			o.Region = c.Region
+		})
+		if err != nil {
+			return err
+		}
+
+		for i := range output.Clusters {
+			res <- output.Clusters[i]
+		}
+
+		if aws.ToString(output.NextToken) == "" {
+			break
+		}
+		config.NextToken = output.NextToken
+	}
+
+	return nil
+}
+func resolveDaxClusterTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	cluster, ok := resource.Item.(types.Cluster)
+	if !ok {
+		return fmt.Errorf("expected types.Cluster but got %T", resource.Item)
+	}
+
+	cl := meta.(*client.Client)
+	svc := cl.Services().DAX
+	response, err := svc.ListTags(ctx, &dax.ListTagsInput{
+		ResourceName: cluster.ClusterName,
+	}, func(options *dax.Options) {
+		options.Region = cl.Region
+	})
+	if err != nil {
+		return err
+	}
+
+	tags := make(map[string]interface{})
+	for _, t := range response.Tags {
+		tags[*t.Key] = t.Value
+	}
+	return resource.Set(c.Name, tags)
+}
+func resolveDaxClusterSecurityGroups(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.Cluster)
+	val := make(map[string]interface{}, len(r.SecurityGroups))
+	for i := range r.SecurityGroups {
+		val[aws.ToString(r.SecurityGroups[i].SecurityGroupIdentifier)] = map[string]interface{}{
```

Comment:
Question: Keep this or make it an array?

Reply:
I would keep it as JSON. btw, why not simply encode `SecurityGroups` as JSON?

Reply:
Since the original types don't have json structs, the resulting keys would not be camel case and they might be very long. `{"SecurityGroupIdentifier": "theidentifier", ...}` etc. But this way we can just have `identifier` instead.